workflow/controller/operator.go

```diff
@@ -1161,6 +1161,13 @@ func (woc *wfOperationCtx) assessNodeStatus(pod *apiv1.Pod, node *wfv1.NodeStatu
 
 	// we only need to update these values if the container transitions to complete
 	if !node.Phase.Fulfilled() && newPhase.Fulfilled() {
+		// If node skipped from pending to complete, send running event now.
```

Comment:
You're assuming (correctly) we missing the running state. Good spot. I think you can make this explicit in the guard condition (i.e. own if statement)

Reply:
Moved to its own statement in https://github.com/argoproj/argo-workflows/pull/5531/commits/b2b5e4f12bdf6ea60f1cd6bf00965d67b04d10ba